### PR TITLE
Assign negative

### PR DESCRIPTION
**Pull request by montegoulding**

### CODE REVIEW

File: engine/src/edittool.cpp [hunk 1]
```diff
@@ -112,7 +113,7 @@ bool MCGradientEditTool::mdown(int2 x, int2 y, uint2 which)
 
 bool MCGradientEditTool::mfocus(int2 x, int2 y)
 {
-	if (m_gradient_edit_point == -1)
+	if (m_gradient_edit_point == MAXUINT4)
```
Comment (livecodeali) [inline engine/src/edittool.cpp]: Presumably this one should be `kInvalidEditPoint` too

Comment (montegoulding) [reply engine/src/edittool.cpp]: Yep will fix thanks

**Comment by livecodeali** [comment]
@livecode-vulcan review ok bf035fd

**Comment by livecodeali** [comment]
Oh, was there a warning that we can enable after this?

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok bf035fdb74d8d95632ab80777a39b42192039d1e

**Comment by livecode-vulcan** [comment]
😎 test success bf035fdb74d8d95632ab80777a39b42192039d1e
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/233)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/237)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/233)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/233)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/237)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/233)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/233)

**Comment by montegoulding** [comment]
I think there is but will require _lots_ lore work before we can `-Werror=conversion`

**Pull request merged**